src/app/user/manager.rs

```diff
@@ -238,9 +239,20 @@ fn hash_password(password: &str) -> Result<String, Error> {
 	if password.is_empty() {
 		return Err(Error::EmptyPassword);
 	}
-	pbkdf2::pbkdf2_simple(password, HASH_ITERATIONS).map_err(|_| Error::Unspecified)
+	let salt = SaltString::generate(&mut OsRng);
+	let result = Pbkdf2
+		.hash_password(password.as_bytes(), &salt)
+		.expect("Invalid generated password salt")
```

Comment:
Could you bubble up the error here instead of panicking?

(I do have to do a pass on all the unspecified errors eventually, but I would like to avoid possible panics)

Reply:
Done